### PR TITLE
Dirty records

### LINKED ISSUE #2: dirty records in characters_metadata
example:
`u5784 +++$+++ WENDY                                              ] +++$+++ m382 +++$+++ happy campers +++$+++ ? +++$+++ ?`

it doesn't show here, but the name field is 52 characters long and record is missing info for gender and credit position.

the record is actually a *duplicate* of the previous record in the file:
`u5783 +++$+++ WENDY +++$+++ m382 +++$+++ happy campers +++$+++ f +++$+++ 2`

i have found two similar cases so far. what is the best way to handle this? changing, i.e. removing a character implies changing all references to it in the other tables. so, remove `u5784` and update all its references to link to `u5783`?

### PR DESCRIPTION
**Pull request by dornheimer**
fixes #2 

resolves references after import and deletes duplicate records.

### CODE REVIEW

**Comment by dornheimer** [comment]
i am slightly confused...the branch was created before the merge, so i updated it with master. i understand that all commits in master appear in the changelog, but why are the commits that i squashed before the merge reappearing?

**Comment by fspoettel** [comment]
Did you branch away from the old branch by any chance? When squashing or merging, the commit hashes change and hence the commits reappear here. This should be fixable with `rebase --onto`, try running `git rebase --onto master dirty_records~2`, solve the merge conflicts and if that fixed it, `git push -f origin dirty_records`

**Edit:** I tried this locally and it worked. Let me know if you need help with this and I can explain it in more detail.

**Comment by dornheimer** [comment]
it worked, thanks. just to make sure i understood everything correctly:
`git rebase --onto master dirty_records~2` reapplies the last two commits from the dirty_records branch on top of master. i don't fully understand what *exactly* `~` does, though.
`git push` has to be forced, because commit history was altered.

**Comment by fspoettel** [comment]
Yes, all correct. `~` is git notation for _start at HEAD of branch and go back n commits_. `--onto` is also pretty useful to get rid of merge commits on a long standing feature branch (it's not uncommon to merge back `master` a few times to get access to new features / fixes etc.)

File: create_db.py [hunk 1]
```diff
@@ -135,6 +135,36 @@ def insert_conversations(session):
     return line_to_conv_mapping
 
 
+def clean_records(session):
+    real_ids = {'u5784': 'u5783', 'u5786': 'u5785', 'u6564': 'u6563'}
+
+    for dup_char_id in ('u5784', 'u5786', 'u6564'):
```
Comment (fspoettel) [inline create_db.py]: You could call [`dict.keys`](https://devdocs.io/python~3.6/library/stdtypes#dict.keys) on `real_ids` here, ensuring that you do not have to hardcode these values twice.

File: create_db.py [hunk 2]
```diff
@@ -135,6 +135,36 @@ def insert_conversations(session):
...
+        real_char_id = real_ids[dup_char_id]
+        dup_char = session.query(Character).get(dup_char_id)
+        real_char = session.query(Character).get(real_char_id)
+
+        char_movie = session.query(Movie).get(dup_char.movie_id)
+        char_movie.characters.remove(dup_char)
+
+        # Make sure there are no duplicate lines
+        char_lines = list(set(dup_char.lines + real_char.lines))
```
Comment (fspoettel) [inline create_db.py:149]: elegant 👍

**Review by fspoettel — COMMENTED** [review]
Small change request, otherwise LGTM

**Review by fspoettel — COMMENTED** [review]
Small comment, otherwise LGTM

**Pull request merged**